majortom_gateway/gateway_api.py

```diff
@@ -45,13 +45,15 @@ def __init__(self, host, gateway_token, ssl_verify=False, basic_auth=None, https
             self.headers['Authorization'] = f'Basic {userAndPass}'
 
     def __build_endpoints(self):
-        if self.https:
-            self.gateway_endpoint = "wss://" + self.host + "/gateway_api/v1.0"
-        else:
+        if self.http:
             self.gateway_endpoint = "ws://" + self.host + "/gateway_api/v1.0"
+        else:
+            self.gateway_endpoint = "wss://" + self.host + "/gateway_api/v1.0"
 
     async def connect(self):
-        if self.https:
+        if self.http:
+            ssl_context = None
+        else:
             ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
 
             if self.ssl_verify:
```

Comment:
This looks like it's really `skip_ssl_verify` since when true it disabled cert checking.

Reply:
Can we ssl verify if we're using http? I thought that was implicit. We actually have a separate if statement (just below there) that if we're using https, checks a separate flag.

Reply:
the http flag is purely to denote if we're using http or https. I assume https unless http is true.